genesis/sensors/base_sensor.py

```diff
@@ -202,6 +287,278 @@ def _get_formatted_data(
+    bias_drift: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    bias_drift_std: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    noise_std: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    jitter_std_in_steps: torch.Tensor = field(
+        default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device)
+    )
+    jitter_in_steps: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    delay_in_steps: torch.Tensor = field(default_factory=lambda: torch.tensor([], dtype=gs.tc_float, device=gs.device))
+    interpolate_for_delay: list[bool] = field(default_factory=list)
+
+
+class NoisySensorBase(RigidSensorBase):
+    """
+    Utility base sensor class for analog sensors that are attached to a RigidEntity.
+    """
+
+    @gs.assert_built
+    def set_bias(self, bias, envs_idx=None):
+        self._set_metadata_tensor(bias, self._shared_metadata.bias, envs_idx, self._cache_size)
+
+    @gs.assert_built
+    def set_bias_drift_std(self, bias_drift_std, envs_idx=None):
+        self._set_metadata_tensor(bias_drift_std, self._shared_metadata.bias_drift_std, envs_idx, self._cache_size)
+
+    @gs.assert_built
+    def set_noise_std(self, noise_std, envs_idx=None):
+        self._set_metadata_tensor(noise_std, self._shared_metadata.noise_std, envs_idx, self._cache_size)
+
+    @gs.assert_built
+    def set_jitter(self, jitter, envs_idx=None):
+        if isinstance(jitter, list):
+            jitter_in_steps = np.array(jitter) / self._manager._sim.dt
+        else:  # scalar or torch tensor or np array
+            jitter_in_steps = jitter / self._manager._sim.dt
+        self._set_metadata_tensor(jitter_in_steps, self._shared_metadata.jitter_std_in_steps, envs_idx, 1)
+
+    @gs.assert_built
+    def set_delay(self, delay, envs_idx=None):
+        self._set_metadata_tensor(delay, self._shared_metadata.delay_in_steps, envs_idx, 1)
+
+    def _set_metadata_tensor(self, input, metadata_tensor, envs_idx=None, n_per_sensor=1):
+        envs_idx = self._sanitize_envs_idx(envs_idx)
+        idx = self._sensor_idx * n_per_sensor
+        metadata_tensor[envs_idx, idx : idx + n_per_sensor] = self._sanitize_for_metadata_tensor(
+            input, shape=(len(envs_idx), n_per_sensor)
+        )
+
+    def _sanitize_for_metadata_tensor(self, input, shape) -> torch.Tensor:
+        if np.isscalar(input):
+            input = [input]
+        tensor_input = torch.tensor(input, dtype=gs.tc_float, device=gs.device)
+        if tensor_input.ndim < len(shape):
+            tensor_input = tensor_input.unsqueeze(0)  # add batch dim
+        if tensor_input.shape[0] == 1 and shape[0] > 1:
+            tensor_input = tensor_input.expand(shape[0], *tensor_input.shape[1:])  # repeat batch dim
```

Comment:
What are you trying to achieve here? You should at least add a comment in the code. Moreover, `tensor_input.shape[0] == 1 and shape[0] > 1` seems to be fragile, but we will see after you write the comment and I can get a clearer understanding of why you are doing this.

Reply:
Ah now I get it. But it is too complicated to read I think. What about:
```python
if tensor_input.ndim == len(shape) - 1:
    # Batch dimension is missing
    tensor_input = tensor_input.unsqueeze(0).expand((shape[0], *tensor_input.shape))
```